### PR TITLE
Clean up code owners

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Cleaning up the code owners

# How

Made the changes discussed in Slack

# Test Plan

None required 

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -18,10 +18,10 @@
 /packages/expo-constants                        @wschurman @esamelson @brentvatne
```
Comment (byCedric) [inline .github/CODEOWNERS:18]:
```suggestion
/packages/expo-constants                        @wschurman @brentvatne
```

Comment (ide) [inline .github/CODEOWNERS:18]: remove these too

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -18,10 +18,10 @@
...
-/packages/expo-dev-launcher                     @lukmccall @esamelson @ajsmth
-/packages/expo-dev-menu                         @lukmccall @esamelson @ajsmth
-/packages/expo-dev-menu-interface               @lukmccall @esamelson @ajsmth
+/packages/expo-dev-client                       @lukmccall @ajsmth
+/packages/expo-dev-launcher                     @lukmccall @ajsmth
+/packages/expo-dev-menu                         @lukmccall @ajsmth
+/packages/expo-dev-menu-interface               @lukmccall @ajsmth
 /packages/expo-device                           @ide @tsapeta
 /packages/expo-document-picker                  @barthap
 /packages/expo-error-recovery                   @lukmccall @esamelson
```
Comment (byCedric) [inline .github/CODEOWNERS:27]:
```suggestion
/packages/expo-error-recovery                   @lukmccall
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -15,16 +15,16 @@
...
+/packages/expo-dev-client                       @lukmccall @ajsmth
+/packages/expo-dev-launcher                     @lukmccall @ajsmth
+/packages/expo-dev-menu                         @lukmccall @ajsmth
+/packages/expo-dev-menu-interface               @lukmccall @ajsmth
 /packages/expo-device                           @ide @tsapeta
 /packages/expo-document-picker                  @barthap
-/packages/expo-error-recovery                   @lukmccall @esamelson
+/packages/expo-error-recovery                   @lukmccall
 /packages/expo-face-detector                    @kudo
 /packages/expo-file-system                      @brentvatne @kudo @esamelson @ide @tsapeta
```
Comment (byCedric) [inline .github/CODEOWNERS:29]:
```suggestion
/packages/expo-file-system                      @brentvatne @kudo @ide @tsapeta
```

File: .github/CODEOWNERS (hunk 1 above)
Comment (byCedric) [reply .github/CODEOWNERS]: Done

**Review by douglowder — APPROVED** [review]

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -15,18 +15,18 @@
...
 /packages/expo-contacts                         @EvanBacon
 /packages/expo-crypto                           @EvanBacon
-/packages/expo-dev-client                       @lukmccall @esamelson @ajsmth
-/packages/expo-dev-launcher                     @lukmccall @esamelson @ajsmth
-/packages/expo-dev-menu                         @lukmccall @esamelson @ajsmth
-/packages/expo-dev-menu-interface               @lukmccall @esamelson @ajsmth
+/packages/expo-dev-client                       @lukmccall @ajsmth
+/packages/expo-dev-launcher                     @lukmccall @ajsmth
+/packages/expo-dev-menu                         @lukmccall @ajsmth
+/packages/expo-dev-menu-interface               @lukmccall @ajsmth
```
Comment (byCedric) [inline .github/CODEOWNERS:24]:
```suggestion
/packages/expo-dev-client                       @lukmccall @douglowder
/packages/expo-dev-launcher                     @lukmccall @douglowder
/packages/expo-dev-menu                         @lukmccall @douglowder
/packages/expo-dev-menu-interface               @lukmccall @douglowder
```

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -68,12 +68,12 @@
 /packages/expo-status-bar                       @brentvatne
 /packages/expo-store-review                     @EvanBacon
 /packages/expo-stories                          @ajsmth
```
Comment (byCedric) [inline .github/CODEOWNERS:70]:
```suggestion
/packages/expo-stories                          @bycedric @EvanBacon
```

**Review by douglowder — APPROVED** [review]

**Pull request merged**